### PR TITLE
Add TextLayer support to Core Animation engine, fix #1488

### LINKED ISSUE #1488: Text Layer(25) not showing while playing animation
## Check these before submitting:
- [] The issue doesn't involve an [Unsupported Feature](https://github.com/airbnb/lottie-ios/blob/master/README.md#supported-after-effects-features)
- [] This issue isn't related to another open issue

## This issue is a:
- [] Non-Crashing Bug (Visual or otherwise)

## Which Version of Lottie are you using?

Lottie 3.2.3

## What Platform are you on? 

- [] iOS

## What Language are you in?
- [] Swift

## Expected Behavior 
Screenshot 2022-01-21 at 10 55 47 PM

## Actual Behavior

Screenshot 2022-01-21 at 10 54 46 PM

## Code Example

 animationView.loopMode = .playOnce
 animationView.play()

## Animation JSON
Will be sharing through mail, it is not allowing to attach json file here

### PR DESCRIPTION
**Pull request by calda**
This PR adds `TextLayer` support to the Core Animation rendering engine.

While working on this, I finally figured out what was causing #1488, which affected both rendering engines since it was an issue in `CoreTextRenderLayer`.

| Before (in both engines) | After (in both engines) |
| ----- | ----- |
| testCoreAnimationRenderingEngine Issues-issue_1488-100 | testCoreAnimationRenderingEngine-1 Issues-issue_1488-100 |

### CODE REVIEW

File: Sources/Private/LayerContainers/Utility/CoreTextRenderLayer.swift [hunk 1]
```diff
@@ -122,12 +123,10 @@ final class TextLayer: CALayer {
 
     ctx.textMatrix = .identity
     ctx.setAllowsAntialiasing(true)
-    ctx.setAllowsFontSmoothing(true)
```
Comment (calda) [inline Sources/Private/LayerContainers/Utility/CoreTextRenderLayer.swift:125]: No clue why, but removing this fixes #1488. All of the text examples in the regression suite still render just fine without this.

**Review by erichoracek — APPROVED** [review]
@calda LGTM! Good stuff

File: Sources/Private/LayerContainers/Utility/CoreTextRenderLayer.swift (hunk 1 above)
Comment (calda) [reply Sources/Private/LayerContainers/Utility/CoreTextRenderLayer.swift:125]: I tested a few different animations with `TextLayer`s in the macOS example app and didn't see any differences with this change.

File: Sources/Private/Experimental/ExperimentalAnimationLayer.swift [hunk 2]
```diff
@@ -57,6 +60,10 @@ final class ExperimentalAnimationLayer: BaseAnimationLayer {
   /// referenced by name in the animation json.
   var imageProvider: AnimationImageProvider
 
+  /// The `FontProvider` that `TextLayer`s use to retrieve the `CTFont`
+  /// that they should use to render their text content
+  var fontProvider: AnimationFontProvider
```
[the remark this answers is not in the document]
Comment (calda) [reply Sources/Private/Experimental/ExperimentalAnimationLayer.swift:65]: Good point, will add this plus a test

[the remark this answers is not in the document]
Comment (calda) [reply Sources/Private/Experimental/ExperimentalAnimationLayer.swift]: https://github.com/airbnb/lottie-ios/pull/1515/commits/52afb9ddef41c18d8bc2b260611185d2a35890da

**Pull request merged**